projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	url        = "http://localhost:8080"
+	maxRetries = 3
+)
+
+func main() {
+	currentTime := time.Now()
```

Comment:
This is maybe _slightly_ early to be getting the current time - ideally you probably want to get the time at around the same time as you get a response.

Imagine you get the current time as 10:00:00, then are told to sleep until 10:00:04, then retry and are told to sleep until 10:00:07 - your current code would sleep for a total of 11 seconds, where you actually only want to sleep for 7 seconds.